test/databases/transaction_database.cpp

```diff
@@ -38,51 +38,502 @@ struct transaction_database_directory_setup_fixture
+    BOOST_REQUIRE(db.create());
+
+    const auto hash1 = tx1.hash();
+    BOOST_REQUIRE(!db.get(hash1));
+
+    const auto hash2 = tx2.hash();
+    BOOST_REQUIRE(!db.get(hash2));
+
+    // Setup end
+    
+    db.store({tx1, tx2}, 100, 110);
+
+    const auto result1 = db.get(hash1);
+    BOOST_REQUIRE(result1);
+    BOOST_REQUIRE(result1.transaction().hash() == hash1);
+
+    const auto result2 = db.get(hash2);
+    BOOST_REQUIRE(result2);
+    BOOST_REQUIRE(result2.transaction().hash() == hash2);
+
+    // Verify computed hash and get via link
+    const auto result3 = db.get(result2.link());
+    BOOST_REQUIRE(result3);
+    BOOST_REQUIRE(result3.transaction().hash() == hash2);
+
+}
+
+BOOST_AUTO_TEST_CASE(transaction_database__store__single_unconfirmed_without_a_block__success)
+{
+    transaction tx1;
+    data_chunk wire_tx1;
+    BOOST_REQUIRE(decode_base16(wire_tx1, TRANSACTION1));
+    BOOST_REQUIRE(tx1.from_data(wire_tx1));
+    
+    const auto path = DIRECTORY "/tx_table";
+    test::create(path);
+    transaction_database db(path, 1000, 50, 0);
+    BOOST_REQUIRE(db.create());
+
+    const auto hash1 = tx1.hash();
+    BOOST_REQUIRE(!db.get(hash1));
+    
+    // Setup end
+    
+    db.store(tx1, 1);
+
+    const auto result1 = db.get(hash1);
+    BOOST_REQUIRE(result1);
+    BOOST_REQUIRE(result1.transaction().hash() == hash1);
+    
+    // Verify computed hash and get via link
+    const auto fetched = db.get(result1.link());
+    BOOST_REQUIRE(fetched);
+    BOOST_REQUIRE(fetched.transaction().hash() == hash1);
+
+}
+
+BOOST_AUTO_TEST_CASE(transaction_database__store__list_unconfirmed_without_a_block__success)
+{
+    transaction tx1;
+    data_chunk wire_tx1;
+    BOOST_REQUIRE(decode_base16(wire_tx1, TRANSACTION1));
+    BOOST_REQUIRE(tx1.from_data(wire_tx1));
+
+    transaction tx2;
+    data_chunk wire_tx2;
+    BOOST_REQUIRE(decode_base16(wire_tx2, TRANSACTION2));
+    BOOST_REQUIRE(tx2.from_data(wire_tx2));
+
+    const auto path = DIRECTORY "/tx_table";
+    test::create(path);
+    transaction_database db(path, 1000, 50, 0);
+    BOOST_REQUIRE(db.create());
+
+    const auto hash1 = tx1.hash();
+    BOOST_REQUIRE(!db.get(hash1));
+
+    const auto hash2 = tx2.hash();
+    BOOST_REQUIRE(!db.get(hash2));
+
+    // Setup end
+    
+    db.store({tx1, tx2});
+
+    const auto result1 = db.get(hash1);
+    BOOST_REQUIRE(result1);
+    BOOST_REQUIRE(result1.transaction().hash() == hash1);
+
+    const auto result2 = db.get(hash2);
+    BOOST_REQUIRE(result2);
+    BOOST_REQUIRE(result2.transaction().hash() == hash2);
+
+    // Verify computed hash and get via link
+    const auto result3 = db.get(result2.link());
+    BOOST_REQUIRE(result3);
+    BOOST_REQUIRE(result3.transaction().hash() == hash2);
+
+}
+
+
+// Candidate/Uncandidate
+// ----------------------------------------------------------------------------
+
+BOOST_AUTO_TEST_CASE(transaction_database__candidate__with_no_input_in_db__returns_false)
```

Comment:
Avoid acronyms (db), don't need to use the term "returns" for the result, just use "false". Really we shouldn't be using db in the tests either. We make a few exceptions for common use terms (like tx). In tests we often just use "instance" for an instance of the object under test.

Reply:
Replaced db with instance in test code.